lightning-net-tokio/src/lib.rs

```diff
@@ -180,6 +180,7 @@ impl Connection {
 					Ok(len) => {
 						prepare_read_write_call!();
 						let read_res = peer_manager.read_event(&mut our_descriptor, &buf[0..len]);
+						peer_manager.process_events();
```

Comment:
This fix isn't sufficient. There is a non-deterministic deadlock on the process_events() PeerHolder lock in the multithreaded test. Two threads can be inside process_events() and the prepare_read_write_call!() has a flag for blocking disconnect with a pending write. I need to learn more about this test and see what the real fix is going to be. Stay tuned.

Reply:
The fix was to call process_events() outside the connection lock so the rl disconnect path can call back into the Connection object and take the lock.